crates/pop-common/src/metadata.rs

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use scale_info::{form::PortableForm, PortableRegistry, Type, TypeDef, TypeDefPrimitive};
+
+/// Formats a specified type, using the registry to output its full type representation.
+///
+/// # Arguments
+/// * `ty`: A reference to the `Type<PortableForm>` to be formatted.
+/// * `registry`: A reference to the `PortableRegistry` to resolve type dependencies.
+pub fn format_type(ty: &Type<PortableForm>, registry: &PortableRegistry) -> String {
```

Comment:
Not a new function, this was moved from the `pop_contract` crate to the `pop_common` crate. However, unit tests have been added for it.